src/components/Network/ModalSelectNetwork.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  availableChainsForToken,
+  getBlockchainsOptions,
+} from "src/features/deposit/components/DepositForm"
+import type { BlockchainEnum } from "src/sdk/poaBridge/constants/blockchains"
+import type {
+  BaseTokenInfo,
+  SupportedChainName,
+  UnifiedTokenInfo,
+} from "src/types/base"
+import { filterChains } from "src/utils/blockchain"
+import { BaseModalDialog } from "../Modal/ModalDialog"
+import { ModalNoResults } from "../Modal/ModalNoResults"
+import { SearchBar } from "../SearchBar"
+import { TooltipInfo } from "../TooltipInfo"
+import { NetworkList } from "./NetworksList"
+
+interface ModalSelectNetworkProps {
+  token: BaseTokenInfo | UnifiedTokenInfo
+  selectNetwork: (network: SupportedChainName) => void
+  selectedNetwork: SupportedChainName | null
+  isOpen?: boolean
+  onClose: () => void
+}
+
+export const ModalSelectNetwork = ({
+  token,
+  selectNetwork,
+  selectedNetwork,
+  isOpen,
+  onClose,
+}: ModalSelectNetworkProps) => {
+  const [searchValue, setSearchValue] = useState("")
+  const chains = getBlockchainsOptions()
+
+  const availableChains = useMemo(() => availableChainsForToken(token), [token])
+  const filteredChains = filterChains(availableChains, searchValue)
+
+  const disabledChains = useMemo(() => {
+    return Object.values(chains).reduce<
+      Record<string, { label: string; icon: ReactNode; value: string }>
+    >((acc, chain) => {
+      if (!filteredChains[chain.value]) {
+        acc[chain.value] = chain
+      }
+      return acc
+    }, {})
+  }, [chains, filteredChains])
+
+  const onChangeNetwork = (network: SupportedChainName) => {
+    selectNetwork(network)
+    onClose()
+  }
+
+  const availableNetworks = Object.keys(filteredChains).map(
+    (key) => key as BlockchainEnum
+  )
+  const disabledNetworks = Object.keys(disabledChains).map(
+    (key) => key as BlockchainEnum
+  )
+
+  return (
+    <BaseModalDialog open={!!isOpen}>
+      <div className="flex flex-col min-h-[680px] md:max-h-[680px] h-full">
+        <div className="z-20 h-auto flex-none -mt-[var(--inset-padding-top)] -mr-[var(--inset-padding-right)] -ml-[var(--inset-padding-left)] px-5 pt-7 pb-4 sticky -top-[var(--inset-padding-top)] bg-gray-1">
+          <div className="flex flex-col gap-4">
+            <div className="flex flex-row justify-between items-center">
+              <Text size="5" weight="bold">
+                Select network
+              </Text>
+              <button type="button" onClick={onClose} className="p-3">
+                <CrossIcon width={18} height={18} />
+              </button>
+            </div>
+            <SearchBar
+              placeholder="Search"
+              query={searchValue}
+              setQuery={setSearchValue}
+            />
+          </div>
+        </div>
+
+        <div className="z-10 flex-1 overflow-y-auto  -mr-[var(--inset-padding-right)] pr-[var(--inset-padding-right)]">
+          {[...availableNetworks, ...disabledNetworks].length === 0 ? (
```

Comment:
```suggestion
          {availableNetworks.length === 0 && disabledNetworks.length === 0 ? (
```

- not creating a new array
- constant time complexity